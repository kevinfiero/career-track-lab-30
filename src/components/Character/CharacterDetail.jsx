import React from 'react';
import PropTypes from 'prop-types';
import './CharacterDetail.css';

const Character = (
  { 
    image, 
    name,
    quote,
    gender,
    personality,
    species,
    birthday,
    phrase,
    skill
  }
) => (
  <section>
    <figure>
      <img className="detailImage" src={image} />
      <figcaption>
        <div>
          <label id="name-label">Name:</label>
          <h1 aria-labelledby="name-label">{name}</h1>
        </div>
        <div>
          <label>Quote:</label>
          <h1>{quote}</h1>
        </div>
        <div>
          <label>Gender:</label>
          <h1>{gender}</h1>
        </div>
        <div>
          <label>Personality:</label>
          <h1>{personality}</h1>
        </div>
        <div>
          <label>Species:</label>
          <h1>{species}</h1>
        </div>
        <div>
          <label>Birthday:</label>
          <h1>{birthday}</h1>
        </div>
        <div>
          <label>Phrase:</label>
          <h1>{phrase}</h1>
        </div>
        <div>
          <label>Skill:</label>
          <h1>{skill}</h1>
        </div>
      </figcaption>
    </figure>
  </section>
);

Character.propTypes = {
  image: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  quote: PropTypes.string.isRequired,
  gender: PropTypes.string.isRequired,
  personality: PropTypes.string.isRequired,
  species: PropTypes.string.isRequired,
  birthday: PropTypes.string.isRequired,
  phrase: PropTypes.string.isRequired,
  skill: PropTypes.string.isRequired
};

export default Character;
