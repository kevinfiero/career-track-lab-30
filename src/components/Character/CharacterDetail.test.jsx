import React from 'react';
import { render, cleanup } from '@testing-library/react';
import CharacterDetail from './CharacterDetail';

describe('CharacterDetail component', () => {
  afterEach(() => cleanup());
  it('renders CharacterDetail', () => {
    const { asFragment } = render(
      <CharacterDetail 
        name = "Agnes" 
        image="https://vignette.wikia.nocookie.net/animalcrossing/images/4/42/Agnes_NH.png/revision/latest?cb=20200804230432"
        quote="You reap what you sow."
        gender="female"
        personality="sisterly"
        species="pig"
        birthday="April 21st (Taurus)"
        phrase="snuffle"
        skill="Cramming for tests"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
