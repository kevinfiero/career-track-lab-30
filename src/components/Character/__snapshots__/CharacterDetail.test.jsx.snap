// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterDetail component renders CharacterDetail 1`] = `
<DocumentFragment>
  <section>
    <figure>
      <img
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/4/42/Agnes_NH.png/revision/latest?cb=20200804230432"
      />
      <figcaption>
        <div>
          <label
            id="name-label"
          >
            Name:
          </label>
          <h1
            aria-labelledby="name-label"
          >
            Agnes
          </h1>
        </div>
        <div>
          <label>
            Quote:
          </label>
          <h1>
            You reap what you sow.
          </h1>
        </div>
        <div>
          <label>
            Gender:
          </label>
          <h1>
            female
          </h1>
        </div>
        <div>
          <label>
            Personality:
          </label>
          <h1>
            sisterly
          </h1>
        </div>
        <div>
          <label>
            Species:
          </label>
          <h1>
            pig
          </h1>
        </div>
        <div>
          <label>
            Birthday:
          </label>
          <h1>
            April 21st (Taurus)
          </h1>
        </div>
        <div>
          <label>
            Phrase:
          </label>
          <h1>
            snuffle
          </h1>
        </div>
        <div>
          <label>
            Skill:
          </label>
          <h1>
            Cramming for tests
          </h1>
        </div>
      </figcaption>
    </figure>
  </section>
</DocumentFragment>
`;
